app/assets/javascripts/admin/addon/components/site-settings/file-types-list.js

```diff
@@ -0,0 +1,76 @@
+import Component from "@ember/component";
+import { action, computed } from "@ember/object";
+import { makeArray } from "discourse-common/lib/helpers";
+
+const IMAGE_TYPES = [
+  "gif",
+  "png",
+  "jpeg",
+  "jpg",
+  "heic",
+  "heif",
+  "webp",
+  "avif",
+  "svg",
+];
+const VIDEO_TYPES = ["mov", "mp4", "webm", "m4v", "3gp", "ogv", "avi", "mpeg"];
+const AUDIO_TYPES = ["mp3", "ogg", "m4a", "wav", "aac", "flac"];
+const DOCUMENT_TYPES = ["txt", "pdf", "doc", "docx", "csv"];
+
+export default class FileTypesList extends Component {
+  tokenSeparator = "|";
+  createdChoices = null;
+
+  @computed("value")
+  get settingValue() {
+    return this.value.toString().split(this.tokenSeparator).filter(Boolean);
+  }
+
+  @computed("settingValue", "setting.choices.[]", "createdChoices.[]")
+  get settingChoices() {
+    return [
+      ...new Set([
+        ...makeArray(this.settingValue),
+        ...makeArray(this.setting.choices),
+        ...makeArray(this.createdChoices),
+      ]),
+    ];
+  }
+
+  @action
+  onChangeListSetting(value) {
+    this.set("value", value.join(this.tokenSeparator));
+  }
+
+  @action
+  onChangeChoices(choices) {
+    this.set("createdChoices", [
+      ...new Set([...makeArray(this.createdChoices), ...makeArray(choices)]),
```

Comment:
Love the use of `Set` to make sure there are only distinct values :+1:

Reply:
Also I didn't know we had `makeArray`; wish JS had the `Array.wrap` ruby equivalent :cry: